src/GraphQl/Resolver/Factory/ItemMutationResolverFactory.php

```diff
@@ -138,15 +143,11 @@ public function __invoke(string $resourceClass = null, string $rootClass = null,
                 }
             }
 
-            if (null !== $item) {
-                $this->validate($item, $info, $resourceMetadata, $operationName);
-                $persistResult = $this->dataPersister->persist($item, $denormalizationContext);
-
-                if (null === $persistResult) {
-                    @trigger_error(sprintf('Returning void from %s::persist() is deprecated since API Platform 2.3 and will not be supported in API Platform 3, an object should always be returned.', DataPersisterInterface::class), E_USER_DEPRECATED);
-                }
+            $this->validate($item, $info, $resourceMetadata, $operationName);
```

Comment:
The previous behavior was: if a null item is returned from the custom mutation, then no persist is done.
With this modification it will not be the case anymore.

Reply:
It should be good now